src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK003SupportAsyncToolWindowAnalyzer.cs

```diff
@@ -83,14 +86,14 @@ private void AnalyzeClassDeclaration(SyntaxNodeAnalysisContext context, INamedTy
             }
 
             INamedTypeSymbol userClassSymbol = context.SemanticModel.GetDeclaredSymbol(declaration, context.CancellationToken);
-            AttributeData packageRegistrationInstance = userClassSymbol?.GetAttributes().FirstOrDefault(a => a.AttributeClass?.Equals(provideToolWindowAttributeType) ?? false);
+            AttributeData? packageRegistrationInstance = userClassSymbol?.GetAttributes().FirstOrDefault(a => a.AttributeClass?.Equals(provideToolWindowAttributeType) ?? false);
             TypedConstant? firstParameter = packageRegistrationInstance?.ConstructorArguments.FirstOrDefault();
             if (firstParameter.HasValue && firstParameter.Value.Kind == TypedConstantKind.Type && firstParameter.Value.Value is INamedTypeSymbol typeOfUserToolWindow)
```

Comment:
Personal preference possibly but it seems more readable to add a check for packageRegistrationInstance!= null in this if condition as well. Could avoid issues with future changes.

Reply:
True. The compiler can't see through enough to know that it has to be non-null for now, so `!` addresses it, but it's fragile to future changes, I agree. I'd like to clean this up, but again, it's kinda out of scope of this PR.